src/components/MediaPlayer/VideoJSPlayer.js

```diff
@@ -89,44 +111,38 @@ function VideoJSPlayer({ isVideo, initStartTime, ...videoJSOptions }) {
     }
 
     if (startTime != null) {
-      player.currentTime(startTime, playerDispatch({ type: 'resetClick' }));
-
       // Mark current timefragment
       if (player.markers) {
         player.markers.removeAll();
         player.markers.add([
           { time: startTime, duration: endTime - startTime, text: 'this' },
         ]);
       }
+      player.currentTime(startTime, playerDispatch({ type: 'resetClick' }));
     }
   }, [startTime, endTime]);
 
-  const handleEnded = (Player) => {
-    // TODO: Need to get this working
-    // if (hasNextSection({ canvasIndex, manifest })) {
-    //   manifestDispatch({ canvasIndex: canvasIndex + 1, type: 'switchCanvas' });
-    //   let newInstance = switchMedia(
-    //     player,
-    //     canvasIndex + 1,
-    //     isPlaying || true,
-    //     captionOn,
-    //     manifest
-    //   );
-    //   playerDispatch({ player: newInstance, type: 'updatePlayer' });
-    //   setCIndex(cIndex + 1);
-    // }
+  const handleEnded = () => {
+    if (hasNextSection({ canvasIndex, manifest })) {
+      manifestDispatch({ canvasIndex: canvasIndex + 1, type: 'switchCanvas' });
+      const oldPlayer = videojs(`videojs-player-${cIndex}`);
```

Comment:
@Dananji Does this create a new instance of a player?  Let's chat on this tomorrow.  I could be wrong, but seems like making multiple calls to `videojs()` is going to create a bunch of different scopes of players?

Reply:
According to the docs [here](https://docs.videojs.com/tutorial-player-workflows.html) it seems to only access the existing players. But as you said there could be side effects from this call.

Reply:
Ah, interesting.